apps/researcher/src/app/[locale]/communities/[id]/page.tsx

```diff
@@ -0,0 +1,99 @@
+import {ChevronLeftIcon} from '@heroicons/react/24/solid';
+import Link from 'next/link';
+import Image from 'next/image';
+import {getTranslations} from 'next-intl/server';
+import {JoinCommunityButton, EditCommunityButton} from './buttons';
+import {getMemberships, getCommunity, isAdminOf} from '@/lib/community';
+
+interface Props {
+  params: {id: string};
+}
+
+export default async function CommunityPage({params}: Props) {
+  const t = await getTranslations('Community');
+
+  const {community, error: communityError} = await getCommunity(params.id);
+  const {memberships, error: membershipsError} = await getMemberships(
+    params.id
```

Comment:
Would it be possible to use a `try...catch...` instead (to use the language's possibilities to its fullest)? For example:

```
try {
  const community = await getCommunity(params.id);
} catch (err) {
  if (err.status === 404) {
    return <div data-testid="no-entity">{t('noEntity')}</div>;
  }
  return <div data-testid="error">{t('error')}</div>;
}
```

An alternative would be to make `community` `undefined` if no community is found (rather than an error): the absence of a community isn't necessarily an error:

```
let community;
try {
 community = await getCommunity(params.id);
} catch (err) {
  return <div data-testid="error">{t('error')}</div>;
}

if (community === undefined) {
  return <div data-testid="no-entity">{t('noEntity')}</div>;
}
```